File: backend/ModuMessenger/src/main/java/com/example/modumessenger/common/data/dto/CommonDataDto.java
package com.example.modumessenger.common.data.dto;

import com.example.modumessenger.common.data.entity.CommonData;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class CommonDataDto {
    private String key;
    private String value;

    public CommonDataDto(String key, String value) {
        setKey(key);
        setValue(value);
    }

    public CommonDataDto(CommonData commonData) {
        setKey(commonData.getKey());
        setValue(commonData.getValue());
    }
}

File: backend/member-service/src/main/kotlin/com/example/memberservice/common/dto/CommonDataDto.kt
package com.example.memberservice.common.dto

import com.example.memberservice.common.entity.CommonData

/** 안드로이드가 기대하는 모양 그대로다: {"key":"version","value":"1.2.3"}. */
data class CommonDataDto(val key: String, val value: String) {
    companion object {
        fun from(commonData: CommonData): CommonDataDto = CommonDataDto(commonData.key, commonData.value)
    }
}
